champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/brute/data/BlockTossObject.java

```diff
@@ -309,15 +320,12 @@ public void impact(Player caster) {
             final EntityProperty relation = nearbyEntry.getValue();
 
             if (relation != EntityProperty.FRIENDLY) {
-                if (!ent.hasLineOfSight(impactLocation)) continue;
                 // Damage anybody who is not friendly
-                damaged.add(ent);
-                Vector knockback = ent.getLocation().toVector().subtract(impactLocation.toVector());
-                final double strength = (radius * radius - ent.getLocation().distanceSquared(impactLocation)) / (radius * radius);
-                VelocityData velocityData = new VelocityData(knockback, strength, false, 0.0, 0.0, 3.0, true);
-                UtilVelocity.velocity(ent, caster, velocityData);
-                UtilDamage.doCustomDamage(new CustomDamageEvent(ent, caster, null, EntityDamageEvent.DamageCause.CUSTOM, damage, false, skill.getName()));
-                UtilMessage.simpleMessage(ent, skill.getName(), "<alt2>%s</alt2> hit you with <alt>%s</alt>.", caster.getName(), skill.getName());
+                if(!ent.hasLineOfSight(impactLocation)) continue;
+                if(!alreadyHit) {
+                    damaged.add(ent);
+                    doDamage(impactLocation, ent, caster, false);
```

Comment:
No point on looping through the entire array if you're only going to hit one entity. You also shouldn't need this loop or the entire logic at all if you did direct impact logic properly to get nearby entities. Unnecessary overhead.

Reply:
The rest of the loop logic doesnt activate unless there are players nearby?

Reply:
fixed anyways